indico/newdle
# Displaying timezone on answers page

## Preview:

<img width="1199" alt="Screen Shot 2020-09-08 at 21 54 43" src="https://user-images.githubusercontent.com/21281337/92542478-3bd5d980-f21f-11ea-9a76-235f3b41ee3a.png">
<img width="1227" alt="Screen Shot 2020-09-02 at 23 37 10" src="https://user-images.githubusercontent.com/21281337/92065040-4272d580-ed75-11ea-9fb5-5e1e769914b3.png">


## Description:

Adding timezone functionality to the newdle answer page as mentioned on #208 . The display timezone used by default is the one obtained from the user's browser. Dates are kept both in the format received by the server and in a timezone sensitive format to show the user

## Todo:

- Active date marked on calendar seems to ignore user timezone, need to look more into it

## Checks:

- All tests pass ✅ 
- All linting rules pass ✅

## Manual testing:

Environment setup:

```
make
source ./.venv/bin/activate
make flask-server
make react-server
```

- Newdle with same timezone as user's has it's dates displayed without changes ✅
- Newdle with option 17:00 - 17:30 on CEST displays as 12:00 - 12:30 on Sao Paulo time ✅
- Newdle displays date correctly with 12 hour difference (e.g. 10:00 August 31st on Japan should be 22:00 August 30th on Sao Paulo) ✅
- Same 12 hour difference check for busy times ✅
- Active date on calendar matches event date correctly when 12 hour difference ✅
- Active date on calendar matches event date correctly when 12 hour difference ✅
- "Accept all options where I'm available" toggle highlights only the aforementioned slots ✅

## Questions:

N/A
